frontend/web/components/metadata/MetadataPage.tsx

```diff
@@ -0,0 +1,351 @@
+import React, { FC, useEffect, useState } from 'react'
+import Button from 'components/base/forms/Button'
+import PanelSearch from 'components/PanelSearch'
+import Icon from 'components/Icon'
+import Panel from 'components/base/grid/Panel'
+import CreateMetadata from 'components/modals/CreateMetadata'
+import { useGetSupportedContentTypeQuery } from 'common/services/useSupportedContentType'
+import Utils from 'common/utils/utils'
+import { ContentType, MetadataModelField } from 'common/types/responses'
+import {
+  useGetMetadataListQuery,
+  useDeleteMetadataMutation,
+} from 'common/services/useMetadata'
+import { useGetMetadataModelFieldListQuery } from 'common/services/useMetadataModelField'
+
+const metadataWidth = [200, 150, 150, 90]
+type MetadataPageType = {
+  organisationId: string
+  projectId: string
+}
+type MergeMetadata = {
+  content_type_fields: MetadataModelField[]
+  id: number
+  name: string
+  type: string
+  description: string
+  organisation: number
+}
+
+const MetadataPage: FC<MetadataPageType> = ({ organisationId, projectId }) => {
+  const [mergeMetadata, setMergeMetadata] = useState<MergeMetadata[]>([])
+  const { data: supportedContentTypes } = useGetSupportedContentTypeQuery({
+    organisation_id: `${organisationId}`,
+  })
+
+  const { data: metadataList } = useGetMetadataListQuery({
+    organisation: organisationId,
+  })
+
+  const { data: MetadataModelFieldList } = useGetMetadataModelFieldListQuery({
+    organisation_id: organisationId,
+  })
+
+  const [deleteMetadata] = useDeleteMetadataMutation()
+
+  let featureContentType: ContentType,
+    segmentContentType: ContentType,
+    environmentContentType: ContentType
+
+  if (supportedContentTypes) {
+    featureContentType = Utils.getContentType(
+      supportedContentTypes,
+      'model',
+      'feature',
+    )
+    segmentContentType = Utils.getContentType(
+      supportedContentTypes,
+      'model',
+      'segment',
+    )
+    environmentContentType = Utils.getContentType(
+      supportedContentTypes,
+      'model',
+      'environment',
+    )
+  }
+
+  useEffect(() => {
+    if (metadataList && MetadataModelFieldList) {
+      const mergeMetadataResult = metadataList?.results.map((item1) => {
+        const matchingItems2 = MetadataModelFieldList?.results.filter(
+          (item2) => item2.field === item1.id,
+        )
+        return {
+          ...item1,
+          content_type_fields: matchingItems2,
+        }
+      })
+
+      setMergeMetadata(mergeMetadataResult)
```

Comment:
doesnt need to be state, can be useMemo

Reply:
Good idea, although perhaps with the new UI/UX design it is no longer necessary.

Reply:
Implemented.